### PR TITLE
Improve the frontend slow query logs.

### PR DESCRIPTION
**Pull request by cyriltovena**
This will help for the log to be more readable, parsable. I've also added the traceID using the context logger.

Signed-off-by: Cyril Tovena <cyril.tovena@gmail.com>

### CODE REVIEW

**Comment by cyriltovena** [comment]
Need rebase on https://github.com/cortexproject/cortex/pull/2521

File: pkg/querier/frontend/frontend.go [hunk 1]
```diff
@@ -152,27 +155,26 @@ func (f *Frontend) Handler() http.Handler {
...
-			"org_id", userID,
-			"url", fmt.Sprintf("http://%s", r.Host+r.RequestURI),
+		logMessage := []interface{}{
+			"msg", "slow query",
+			"host", r.Host,
+			"path", r.URL.Path,
 			"time_taken", queryResponseTime.String(),
 		}
+		for k, v := range r.URL.Query() {
+			append(logMessage, fmt.Sprintf("qs_%s", k), strings.Join(v, ","))
```
Comment (pracucci) [inline pkg/querier/frontend/frontend.go:171]: There's a bug here. `logMessage` is not updated with the result of `append()`. I fix it for you.

**Comment by pracucci** [comment]
For other reviewers, this is an example log:
```
level=info ts=2020-04-24T05:41:30.1960564Z caller=frontend.go:177 org_id=fake trace_id=565b3488fe9aa3a6 msg="slow query" host=host.docker.internal:8007 path=/api/prom/api/v1/query_range time_taken=10.1883ms qs_start=1587703290 qs_end=1587706890 qs_step=15 qs_query="up "
```

File: pkg/querier/frontend/frontend.go [hunk 2]
```diff
@@ -152,27 +155,26 @@ func (f *Frontend) Handler() http.Handler {
...
 		}
+		for k, v := range r.URL.Query() {
+			logMessage = append(logMessage, fmt.Sprintf("qs_%s", k), strings.Join(v, ","))
+		}
 		pf := r.PostForm.Encode()
 		if pf != "" {
 			logMessage = append(logMessage, "body", pf)
 		}
-		level.Info(f.log).Log(logMessage...)
+		level.Info(util.WithContext(r.Context(), util.Logger)).Log(logMessage...)
```
Comment (pstibrany) [inline pkg/querier/frontend/frontend.go:177]:
```suggestion
		level.Info(util.WithContext(r.Context(), f.log)).Log(logMessage...)
```

**Pull request merged**